tests/tests/test_demo_artifact.py

```diff
@@ -55,14 +55,15 @@ def run_demo_script_up():
                     conftest.docker_compose_instance, 'up'
                 ],
                 cwd="..",
+                stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
                 env=test_env)
             logging.info('Started the demo script')
             password = ""
             time.sleep(60)
             for line in iter(proc.stdout.readline, ''):
                 logging.info(line)
-                if "Login password:" in line.strip():
+                if exit_cond in line.strip():
```

Comment:
If the `exit_cond` is anything but "Login password:", the code block below will fail, right?

Reply:
It will block on readline, until the test times out.

Reply:
So you are saying that for test `demo_up_down_up` there is no really `exit_cond` because it will block in readline * before* getting here?

The code is not very descriptive, if you pass an exit_cond that will never reach...

Reply:
It has two different exit conditions. One for the first case. Bringing it up the first time, in which it has to provide a password. 
Then in the second case, the exit condition is the message that the user does not exist.

Reply:
You mean that the user already exists, right?

So back to my original point: in the case of `exit_cond` being anything but "Login password": how will the following code work?
```
password = line[-13:-1]
...
```

Reply:
I mean, I know "how" it works, because you only check for the len, so no matter if the password is a sane one or just some random substring from "The user already exists"...

My point is that we are using this `exit_cond` for two things, as a proper exit condition and as an expectation for where to grab the password from.

Reply:
Ahh, yeh. ur right! Sry, did not see this. Fixing!

Reply:
How about now?

Reply:
Good! :+1: